shell/testing/observatory/test.dart

```diff
@@ -98,51 +98,53 @@ Future<Null> testHttpAssetRequest(Uri uri) async {
 Future<Null> testStartPaused(Uri uri) async {
   uri = uri.replace(scheme: 'ws', path: 'ws');
   final WebSocket webSocketClient = await WebSocket.connect(uri.toString());
-  final ServiceClient serviceClient = new ServiceClient(webSocketClient);
+  final Completer<dynamic> isolateStartedId = new Completer<dynamic>();
+  final Completer<dynamic> isolateRunnableId = new Completer<dynamic>();
+  final Completer<dynamic> isolateResumeId = new Completer<dynamic>();
+  final ServiceClient serviceClient = new ServiceClient(webSocketClient,
+      isolateStartedId: isolateStartedId,
+      isolateRunnableId: isolateRunnableId,
+      isolateResumeId: isolateResumeId);
+  await serviceClient.invokeRPC('streamListen', <String, String>{ 'streamId': 'Isolate'});
+  await serviceClient.invokeRPC('streamListen', <String, String>{ 'streamId': 'Debug'});
 
-  // Wait until we have the isolateId.
+  final Map<String, dynamic> response = await serviceClient.invokeRPC('getVM');
+  Expect.equals(response['type'], 'VM');
   String isolateId;
-  while (isolateId == null) {
-    final Map<String, dynamic> response = await serviceClient.invokeRPC('getVM');
-    Expect.equals(response['type'], 'VM');
-    if (response['isolates'].length > 0) {
-      isolateId = response['isolates'][0]['id'];
-    }
+  if (response['isolates'].length > 0) {
+    isolateId = response['isolates'][0]['id'];
+  } else {
+    // Wait until isolate starts.
+    isolateId = await isolateStartedId.future;
   }
 
   // Grab the isolate.
-  Map<String, dynamic> isolate;
-  while(true) {
+  Map<String, dynamic> isolate = await serviceClient.invokeRPC('getIsolate', <String, String>{
+      'isolateId': isolateId,
+  });
+  Expect.equals(isolate['type'], 'Isolate');
+  Expect.isNotNull(isolate['pauseEvent']);
+  // If it is not runnable, wait until it becomes runnable.
```

Comment:
The code here is interested in the isolate being paused at start, so it should look for that condition instead of runnable.

Reply:
Done.